ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/textures.cpp

```diff
@@ -29,6 +29,22 @@
 
 namespace enigma {
 
+// Because GLES won't do the conversion for us
+unsigned char* bgra2rgba(unsigned char* in, int w, int h) {
+  unsigned char* out = new unsigned char[w * h * 4];
+  int offset = 0;
+  for (int y = 0; y < h; ++y) {
+    for (int x = 0; x < w; ++x) {
+      out[offset] = in[offset + 2];
+      out[offset + 1] = in[offset + 1];
+      out[offset + 2] = in[offset];
+      out[offset + 3] = in[offset + 3];
+      offset += 4;
+    }
+  }
+  return out;
+}
+
```

Comment:
Are you using GLAD or what? I remember what I think we did to solve the BGRA problem before, we used the extension which was supported on like 98% of hardware according to some Khronos statistics.
https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_texture_format_BGRA8888.txt

Edit: Correction* it's supported by 99.2% of devices.
https://opengles.gpuinfo.org/listextensions.php

![BGRA GLES EXT Support](https://user-images.githubusercontent.com/3212801/76181923-409ad180-6199-11ea-91b4-69355b1daaab.png)

Reply:
If i put bgra & bgra everything blue and red are swapped even on desktop. The real solution is to just kill this bgra nonsense in the engine but thats far too much work for me. Right now you have opengl doing the conversion but gles cannot do that conversion even if it supports bgra.

Reply:
No I mean put it back the way that it was (BGRA & **_RGBA_**) and add the extension. The desktop was obviously fine before you changed _anything_! I recall API tracing GMS and seeing that it was in fact still using BGRA internally and I also recall finding somewhere that it too used this extension.

Reply:
See GameMaker discussion as well:
https://forum.yoyogames.com/index.php?threads/switching-gms2-from-bgr-to-standard-rgb.33526/

Reply:
The extension has nothing todo w/ conversion. bgra is already enable but GLES will not convert bgra to rgba regardless of any extension

Reply:
Try GL_BGRA_EXT

Reply:
Also TKG made a helper that appears to do the same thing:
https://github.com/enigma-dev/enigma-dev/pull/1904/files#diff-1246ecb7fd9dc46a9b17aa578b0d5c0bR37

Reply:
I think Sam just checked this method in as part of his icon PR. Please try to reuse that.

Reply:
fundies is also being difficult and I don't know whether to believe him about BGRA.

Reply:
Here's that thing tkg added.
https://github.com/enigma-dev/enigma-dev/blob/2d1220f23fa6dd057bf478eba8f25e5a82d34b6d/ENIGMAsystem/SHELL/Universal_System/image_formats.cpp#L188